.github/actions/go-e2e/action.yaml

```diff
@@ -0,0 +1,55 @@
+name: 'Go E2E Tests Reusable Action'
+description: 'Run Go E2E tests using the provided TAS server and Go version.'
+inputs:
+  tas_address:
+    description: 'TAS server address'
+    required: true
+    type: string
+  go_version:
+    description: 'Go version'
+    required: true
+    default: '1.21'
+    type: string
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout go test source code
+      uses: actions/checkout@v4
+      with:
+        repository: "securesign/sigstore-e2e"
+        path: e2e
+    - name: Install Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ inputs.go_version }}
+
+    - name: Resolve TAS_SINGLENODE hostname
+      shell: bash
+      run: |
+        echo "${{ inputs.tas_address }} fulcio.myrhtas tuf.myrhtas rekor.myrhtas tsa.myrhtas rekor-search.myrhtas cli-server.myrhtas dex-idp" | sudo tee -a /etc/hosts
+
+    - name: Resolve certificates
+      shell: bash
+      run: |
+        openssl s_client -showcerts -connect rekor.myrhtas:443 < /dev/null | awk '/BEGIN CERTIFICATE/,/END CERTIFICATE/{ if(/BEGIN CERTIFICATE/){a++}; out="cert"a".pem"; print >out}'
```

Comment:
We do a lot of the same stuff as you do here in https://github.com/securesign/artifact-signer-ansible/blob/main/vm-testing/test/test-sign-blob.sh#L7 - could you possibly try to extract the common parts of the referenced file and your script to a common shared file and reuse it here? It'd be nice to prevent the code duplication.

Reply:
Good point! Finally, after some suffering time, I was able to reuse the same function in both scenarios.